src/Hintman/HLint.hs

```diff
@@ -92,28 +108,62 @@ createFileDownloadUrl PrInfo{..} (toText -> file) = T.intercalate "/"
     , file
     ]
 
-{- | Creates the comment text from the HLint 'Idea's.
+-- | Create a 'Comment' from all necessary data.
+createComment :: FileDeltas -> FilePath -> Text -> Idea -> Maybe Comment
+createComment _fd (toText -> commentPath) content idea =
+    createCommentText content idea >>= \commentBody -> Just Comment
+        { commentPosition = 0 -- TODO: get this from line number
+        , ..
+        }
+
+{- | Creates the comment text from the HLint 'Idea's and file content.
 -}
-createComment :: Idea -> Maybe Text
-createComment Idea{..} = ideaTo >>= \to ->
+createCommentText :: Text -> Idea -> Maybe Text
+createCommentText content Idea{..} = ideaTo >>= \to ->
     if ideaSeverity == Ignore
     then Nothing
-    else Just $ unlines $
+    else buildWholeLine to >>= \newLine -> Just $ unlines $
```

Comment:
We can rewrite this code in a slightly cleaner way using `do`-notation for `Maybe` if you think it's better:

```haskell
createCommentText content Idea{..} = do
    to <- ideaTo
    guard $ ideaSeverity /= Ignore
    newLine <- buildWholeLine to
    ...
```

Reply:
Nice! Much cleaner!